src/app/fdctl/run/tiles/fd_replay.c

```diff
@@ -651,6 +671,32 @@ publish_account_notifications( fd_replay_tile_ctx_t * ctx,
   FD_LOG_DEBUG(("TIMING: notify_account_time - slot: %lu, elapsed: %6.6f ms", curr_slot, (double)notify_time_ns * 1e-6));
 }
 
+static void
+replay_plugin_publish( fd_replay_tile_ctx_t * ctx,
+                       ulong sig,
+                       uchar const * data,
+                       ulong data_sz ) {
+  uchar * dst = (uchar *)fd_chunk_to_laddr( ctx->replay_plugin_out_mem, ctx->replay_plugin_out_chunk );
+  fd_memcpy( dst, data, data_sz );
+  ulong tspub = (ulong)fd_frag_meta_ts_comp( fd_tickcount() );
+  fd_mcache_publish( ctx->replay_plugin_out_mcache, ctx->replay_plugin_out_depth, ctx->replay_plugin_out_seq, sig, ctx->replay_plugin_out_chunk, data_sz, 0UL, 0UL, tspub );
```

Comment:
Use `fd_stem_publish` here, so you get flow control. Then you can remove `replay_plugin_out_mcache`, `replay_plugin_out_sync`, `replay_plugin_out_depth` and `replay_plugin_out_seq`

Reply:
when i use stem_publish, I get blocked in flow control. one could argue that flow control is wrong here anyway.

Reply:
i got this to work

Reply:
i got this to work